src/libraries/System.Text.Json/tests/Serialization/ReferenceHandlerTests.IgnoreCycle.cs

```diff
@@ -0,0 +1,324 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public class ReferenceHandlerTests_IgnoreCycle
+    {
+        private static readonly JsonSerializerOptions s_optionsIgnoreCycles =
+            new JsonSerializerOptions { ReferenceHandler = ReferenceHandler.IgnoreCycle };
+
+        [Fact]
+        public async Task IgnoreCycles_OnObject()
```

Comment:
For this and the following collection-related types, I notice that the element types are all `object`. Do we have any with strongly typed class-type, value-type, and nullable-value-type elements?

Reply:
I added more types that contain a property with object or a concrete type. It should cover boxed classes, non-boxed classes and boxed structs.

I am not testing non-boxed value types since those are ignored/never tracked in the stack of references and `Nullable<T>` can't be boxed.